modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -88,19 +109,37 @@ const onSubmitCellForm = (editor: Editor, cells: HTMLTableCellElement[], api) =>
   });
 };
 
+const getData = (editor: Editor, startCell: SugarElement<Element>, cells: SugarElement<HTMLTableCellElement>[]) => {
+  const table = TableLookup.table(startCell).getOrNull();
+
+  const warehouse = Warehouse.Warehouse.fromTable(table);
+
+  const columns = Warehouse.Warehouse.justColumns(warehouse);
+  const allCells = Warehouse.Warehouse.justCells(warehouse);
+
+  const selectedCells = Arr.filter(allCells, (cellA) =>
+    Arr.exists(cells, (cellB) =>
+      cellA.element.dom === cellB.dom
+    )
+  );
+
+  const cellsData: CellData[] = Arr.map(selectedCells, (cell) =>
+    Helpers.extractDataFromCellElement(editor, cell.element.dom, columns[cell.column], hasAdvancedCellTab(editor))
```

Comment:
This won't work if something is using a colspan, as the array returned from `justColumns` isn't indexed by column number.

Reply:
It might be worthwhile to add a new function to `Warehouse` similar to `getAt` but for the column, instead of the cell.